### PR TITLE
refactor: Fix error handling msg in flobject.

### PR DESCRIPTION
**Pull request by prmukherj**
Improve error handling message.

### CODE REVIEW

File: src/ansys/fluent/core/filereader/case_file.py [hunk 1]
```diff
@@ -192,7 +192,9 @@ def __call__(self, name: str = ""):
         except KeyError:
             raise ValueError(
                 allowed_name_error_message(
-                    "config-vars", name, list(self._variables.keys())
+                    context="config-vars",
+                    trial_name=name,
+                    allowed_values=list(self._variables.keys()),
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/filereader/case_file.py:197]:
```suggestion
                    allowed_values=list(self._variables),
```

File: src/ansys/fluent/core/solver/flobject.py [hunk 2]
```diff
@@ -1015,7 +1019,9 @@ def __getattr__(self, name: str):
         except KeyError:
             raise AttributeError(
                 allowed_name_error_message(
-                    "Settings objects", name, self.get_active_child_names()
+                    context="Settings objects",
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flobject.py:1022]: Probably, this should be raising _from_.

File: src/ansys/fluent/core/solver/error_message.py [hunk 3]
```diff
@@ -10,11 +10,16 @@ def closest_allowed_names(trial_name: str, allowed_names: str) -> List[str]:
...
+    allowed_values: Any,
+    context: Optional[str] = None,
+    trial_name: Optional[str] = None,
+    message: Optional[str] = None,
 ) -> str:
     """Provide an error message with the closest names matching the 'trial_name' from
     the 'allowed_values' list."""
-    message = f"{trial_name} is not an allowed {context} name.\n"
+    if not message:
+        message = f"{trial_name} is not an allowed {context} name"
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/error_message.py:21]: Why only generate the message conditionally? This is the whole point if this function.

File: src/ansys/fluent/core/solver/flobject.py (hunk 2 above)
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flobject.py:1022]: Does this generate f"{trial_name} is not an allowed Settings objects name"?

File: src/ansys/fluent/core/solver/flobject.py [hunk 4]
```diff
@@ -1202,7 +1208,9 @@ def __getitem__(self, name: str) -> ChildTypeT:
                 )
             raise KeyError(
                 allowed_name_error_message(
-                    "Settings objects", name, self.get_object_names()
+                    context="Settings objects",
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/flobject.py:1211]: same question here

File: src/ansys/fluent/core/solver/flobject.py (hunk 2 above)
Comment (prmukherj) [reply src/ansys/fluent/core/solver/flobject.py:1022]: yes

File: src/ansys/fluent/core/solver/error_message.py (hunk 3 above)
Comment (prmukherj) [inline src/ansys/fluent/core/solver/error_message.py:21]:
```suggestion
        message = f"'{context}' has no attribute '{trial_name}'."
```

**Review by hpohekar — APPROVED** [review]

File: src/ansys/fluent/core/solver/error_message.py [hunk 5]
```diff
@@ -10,11 +10,16 @@ def closest_allowed_names(trial_name: str, allowed_names: str) -> List[str]:
...
+    allowed_values: Any,
+    context: Optional[str] = None,
+    trial_name: Optional[str] = None,
+    message: Optional[str] = None,
 ) -> str:
     """Provide an error message with the closest names matching the 'trial_name' from
     the 'allowed_values' list."""
-    message = f"{trial_name} is not an allowed {context} name.\n"
+    if not message:
+        message = f"'{context}' has no attribute '{trial_name}'"
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/solver/error_message.py:21]: We are still not spot on here. E.g., _Settings objects has_. So in those cases we should send the object name, equivalent to the other flobject scenarios.

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**